docs/api/menu.md

```diff
@@ -257,28 +251,62 @@ Menu.setApplicationMenu(menu)
 
 ### Render process
 
-Below is an example of creating a menu dynamically in a web page
-(render process) by using the [`remote`](remote.md) module, and showing it when
-the user right clicks the page:
+To create menus initiated by the renderer process, send the required
+information to the main process using IPC and have the main process display the
+menu on behalf of the renderer.
+
+Below is an example of creating a menu with dynamic information in a web page,
+and showing it when the user right clicks the page:
 
 ```html
 <!-- index.html -->
 <script>
-const { remote } = require('electron')
-const { Menu, MenuItem } = remote
-
-const menu = new Menu()
-menu.append(new MenuItem({ label: 'MenuItem1', click() { console.log('item 1 clicked') } }))
-menu.append(new MenuItem({ type: 'separator' }))
-menu.append(new MenuItem({ label: 'MenuItem2', type: 'checkbox', checked: true }))
+const { ipcRenderer } = require('electron')
 
 window.addEventListener('contextmenu', (e) => {
   e.preventDefault()
-  menu.popup({ window: remote.getCurrentWindow() })
-}, false)
+  const template = [
+    { label: 'Menu Item 1', id: 'menu-item-1' },
+    { type: 'separator' },
+    { label: 'Menu Item 2', type: 'checkbox', checked: true, id: 'menu-item-2' },
+  ]
+  ipcRenderer.invoke('show-menu', template).then((clickedId) => {
+    if (clickedId === 'menu-item-1') {
+      console.log('item 1 clicked')
+    }
+  })
+})
 </script>
 ```
 
+```js
+// main.js
+const { BrowserWindow, ipcMain, Menu } = require('electron')
+
+ipcMain.handle('show-menu', (event, template) => {
+  const menu = Menu.buildFromTemplate(template)
+  menu.popup(BrowserWindow.fromWebContents(event.sender))
+  return new Promise((resolve, reject) => {
```

Comment:
This is kinda nasty, is there a better way of doing this over IPC? It's a common use case and this isn't super ideal 🤔  We should think about a nice way of doing this before recommending this particular technique.

Reply:
I'm open to suggestions, this is the best I've thought of so far.

See the PR description for proposed changes to the `Menu` API that would simplify this.

Reply:
How about using a simple example like this:

```js
// renderer
window.addEventListener('contextmenu', (e) => {
   e.preventDefault()
   ipcRenderer.send('show-context-menu')
 })

ipcRenderer.on('context-menu-command', () => { ... })

// main
ipcMain.on('show-context-menu', (event) => {
  const template = [
    { label: 'Menu Item 1', click: () => { event.sender.send('context-menu-command') } },
    { type: 'separator' },
    { label: 'Menu Item 2', type: 'checkbox', checked: true },
  ]
  const menu = Menu.buildFromTemplate(template)
  menu.popup(BrowserWindow.fromWebContents(event.sender))
})
```

The purpose of the example is to show how to show menu from renderer process, so a ping-pong style example could probably serve the purpose best.

Reply:
@zcbenz seems reasonable. Updated.